Moved AWS Lambda docs from Notion to develop docs (#15034)
getsentry/sentry-docs: develop-docs/sdk/platform-specifics/serverless-sdks/aws-lambda-development-environment.mdx

---
title: AWS Lambda Development Environment
sidebar_order: 200
---

> This guide will explain how you can setup a development environment to work on the [AWS Lambda Integration](https://docs.sentry.io/product/integrations/cloud-monitoring/aws-lambda/). The development environment is a local instance of `sentry` and configuration in AWS to make everything work together.

Reviewer: ```suggestion
This guide will explain how you can setup a development environment to work on the [AWS Lambda Integration](https://docs.sentry.io/product/integrations/cloud-monitoring/aws-lambda/). The setup includes running a local instance of `sentry` and configuring AWS so that both environments work together.
```

<Alert level="warning" title="IMPORTANT">

**This guide is only if you want to send data from an AWS Lambda function to your local `sentry` instance.**

When working on the Sentry AWS Lambda layer from one of the SDKs, the workflow is that you have your example Lambda function in a dev account on AWS and then deploy your local layer to the dev account and attach it to your example function. How do to this in Python is described in the [contribution guide](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md#contributing-to-sentry-aws-lambda-layer) of the Python SDK.

</Alert>

## Configuration of AWS

To emulate a Sentry integration with a project base on AWS Lambda functions you need:

- One AWS account representing Sentry having an IAM and a S3 Bucket containing a JSON config file.
- One AWS account representing the user that has a example Lambda function.

> ### What You Need in Detail

Reviewer 2: This section reads a little jumpy. Some steps are telling you to do something, while others take a passive tone. It makes it hard for me to keep track of what I'm reading about from one step to the next. 

Example: "we assume the AWS account ID is..." vs "create an **IAM user**..." vs "a A **S3 Bucket in “Sentry Account”**..."

I wonder if this section gets turned into more of a step-by-step of all the configurations? I like "do" actions like _create_ vs passive listing like _a thing_, _we assume you have xyz_, etc

Reviewer 3: I much prefer starting with a verb as well.

I turned the first bullet point into a free-standing paragraph, and converted the remaining ones to start with a verb.

> ### What You Need in Detail

Reviewer 2: ```suggestion
### Detailed Configuration Steps
```

- For the **“Sentry Account”** in AWS we assume the AWS Account ID is `1111 1111 1111` in this guide. This account can be the shared official dev AWS account of Sentry, or you can create a personal one (credit card required for this)
- Create an **IAM user** in the **“Sentry account”** that can create S3 buckets, and has permission to assume roles. The policy that can be directly attached to the IAM user looks like this:

>   _TODO: check if we can restrict those permissions more_

Reviewer 2: @antonpirker - I'm assuming this still needs to be checked? Just need to make sure to take this out/modify it before publishing

  ```json
  {
    "Version": "2012-10-17",
    "Statement": [
      {
        "Sid": "VisualEditor0",
        "Effect": "Allow",
        "Action": "sts:AssumeRole",
        "Resource": "*"
      }
    ]
  }
  ```

- A **S3 Bucket in “Sentry Account”** that is accessible by the public to host the CloudFormation configuration file. (more on this later). The S3 Bucket Policy should look like this (in this example “sentry-dev-cloudformation” is the name of the S3 bucket):

>   _TODO: check if we can restrict those permissions more_

Reviewer 2: Just another call-out on changing this line before publishing.

Reviewer 3: Same here, turned into a comment.

- `aws-lambda.python.layer-version` same as above but for Python based Lambda functions.

## Configuration of Ngrok

With [Ngrok](https://ngrok.com/) you get an URL that points to your local computer. So everyone on the internet can talk to your Sentry installation on your computer.

This is needed so your AWS Lambda function can send its errors/tracing to your local Sentry installation.

Follow the instructions on the [Ngrok documentation](https://develop.sentry.dev/development/ngrok/) page to install ngrok.

Now start Ngrok like the following:

```bash
ngrok http 8000
```

If ngrok starts it outputs the URL that your computer is now available at. Copy the HTTP URL (it should look something like `http://xxxx-xxx-xxx-x-xxx.ngrok.io`).

If you follow the guide in the ngrok page linked above you will get access to the Sentry Ngrok account and you can create a subdomain that is custom and not always changing when you restart ngrok. This is highly recommended.

You now have to tell your sentry installation its new URL by adding the following line to your `~/.sentry/config.yml` :

```yaml
system.url-prefix: "http://xxxx-xxx-xxx-x-xxx.ngrok.io"
```

> Make sure to restart all the development environment, to make sure everyone knows about the new URL. (with `devservices down && devservices up`)

Reviewer 3: Changed to "all services", I agree that it's part of the stack.

## Start the Local Sentry Server

> If you now run your local Sentry with this command and it will have all the information it needs:

Reviewer 3: Addressed, thanks!

```bash
devservices serve
```

> Now everything is ready for serverless integration development.

Reviewer 3: Good point but I do not have the full context and @antonpirker has unfortunately left.

The description below the title says the guide is for "running a local instance of `sentry` and configuring AWS", but I agree that it would be ideal to elaborate on when this is useful.
